src/Illuminate/Mail/Mailable.php

```diff
@@ -872,21 +872,27 @@ public function with($key, $value = null)
      * @param  array  $options
      * @return $this
      */
-    public function attach($file, array $options = [])
+    public function attach($files, array $options = [])
     {
-        if ($file instanceof Attachable) {
-            $file = $file->toMailAttachment();
-        }
-
-        if ($file instanceof Attachment) {
-            return $file->attachTo($this);
+        if(!is_iterable($files)) {
+            $files = explode(',', $files);
```

Comment:
This doesn't work when `$files` is an attachable. (Tests are failing.) You'll also need to add some tests for the new behavior.